### ISSUE TITLE
Invoking native executable on Linux with empty array result in "null/empty" parameter

### ISSUE THREAD

**Issue by ggirard07** [PowerShell/PowerShell#21194]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Invoking following command on Linux (Ubuntu 22) result in error as AzureCLI receives an extra "empty" parameter.
```pwsh
PS /home/me> $foo = "".Split()
PS /home/me> az account show $foo --debug
cli.knack.cli: Command arguments: ['account', 'show', '', '--debug']
```

This used to work fine in `7.2.x` on Ubuntu 22
This still works fine on Windows 10

I use this technique to include optional parameter when constructing command line based on conditions.

### Expected behavior

```console
PS /home/me> $foo = "".Split()
PS /home/me> az account show $foo --debug
cli.knack.cli: Command arguments: ['account', 'show', '--debug']
```

### Actual behavior

```console
PS /home/me> $foo = "".Split()
PS /home/me> az account show $foo --debug
cli.knack.cli: Command arguments: ['account', 'show', '', '--debug']
```

### Environment data

```powershell
PS /home/me> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.3.4
PSEdition                      Core
GitCommitId                    7.3.4
OS                             Linux 5.15.133.1-microsoft-standard-WSL2 #1 SMP Thu Oct 5 21:02:42 UTC 2023
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by mklement0** [comment]
It is the old behavior that was broken: `''` is a legitimate argument that should be passed as such.

This is fixed in v7.3+, but the old behavior - which notably also is broken with respect to passing _embedded `"` - is still available on an opt-in basis, by (temporarily) setting the [`$PSNativeCommandArgumentPassing` preference variable](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_Preference_Variables#psnativecommandargumentpassing)  to `'Legacy'`.

**Comment by ggirard07** [comment]
Thanks, I looked only at the breaking changes in the 7.3/7.4 release notes. Sorry for the noise.
